src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -601,15 +606,66 @@ def build_binary(
         }
         api_calls = [*select_left.api_calls, *select_right.api_calls]
 
+        # This is a temporary workaround for query comparison. The previous CTE optimization
```

Comment:
You mean this? https://github.com/snowflakedb/snowpark-python/blob/89913c27ed588e0f49ba729cb9f85b835a17bf2d/src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py#L1456

this is not introduced by CTE, and it's used for retrieving result from previous query id https://github.com/snowflakedb/snowpark-python/blob/89913c27ed588e0f49ba729cb9f85b835a17bf2d/src/snowflake/snowpark/_internal/server_connection.py#L644-L646

you can follow this https://github.com/snowflakedb/snowpark-python/blob/89913c27ed588e0f49ba729cb9f85b835a17bf2d/src/snowflake/snowpark/_internal/analyzer/cte_utils.py#L148 to compare sql query text and query_params. Maybe you can implement __eq__ in Query so you directly use `if current_query in existing_queries`

Reply:
sure, updated